### PR TITLE
Pesomka/client cred silent

### PR DESCRIPTION
**Pull request by SomkaPe**
- silent flow for client credential grant 
- use tenant id from request authority if id token not returned
- use  AutoCloseable  for cache Aspect application

### CODE REVIEW

File: src/integrationtest/java/com.microsoft.aad.msal4j/ClientCredentialsIT.java [hunk 1]
```diff
@@ -72,8 +72,14 @@ private void assertAcquireTokenCommon(String clientId, IClientCredential credent
...
-        // TODO AuthenticationResult should have an getAccountInfo API
-        // Assert.assertEquals(labResponse.getUser().getUpn(), result.getAccountInfo().getUsername());
+
+        result = cca.acquireTokenSilently(SilentParameters
+                .builder(Collections.singleton(KEYVAULT_DEFAULT_SCOPE))
+                .build())
+                .get();
+
+        Assert.assertNotNull(result);
+        Assert.assertNotNull(result.accessToken());
```
Comment (sangonzal) [inline src/integrationtest/java/com.microsoft.aad.msal4j/ClientCredentialsIT.java]: Might also be worth writing a test that has multiple tokens in cache for same clientId and tenant but different scopes and check that the right token is returned.

File: src/main/java/com/microsoft/aad/msal4j/TokenRequest.java [hunk 2]
```diff
@@ -51,12 +52,15 @@
...
+    private final URL tokenEndpointUrl;
     private final MsalRequest msalRequest;
     private final ServiceBundle serviceBundle;
 
-    TokenRequest(final URL url, MsalRequest msalRequest, final ServiceBundle serviceBundle) {
-        this.url = url;
+    TokenRequest(Authority requestAuthority, MsalRequest msalRequest, ServiceBundle serviceBundle)
+            throws MalformedURLException {
+        this.requestAuthority = requestAuthority;
+        this.tokenEndpointUrl = new URL(requestAuthority.tokenEndpoint());
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/TokenRequest.java]: Do you need both here, Perhaps authority.tokenEndpointUrl could do this instead?

Comment (SomkaPe) [reply src/main/java/com/microsoft/aad/msal4j/TokenRequest.java]: good point, changed

File: src/integrationtest/java/com.microsoft.aad.msal4j/ClientCredentialsIT.java (hunk 1 above)
Comment (SomkaPe) [reply src/integrationtest/java/com.microsoft.aad.msal4j/ClientCredentialsIT.java]: good point - extended to few scopes and found serious bug

**Pull request merged**